### PR TITLE
fix hardhat & add foundry examples

### PR DESCRIPTION
**Pull request by mnedelchev-vn**
Hi @m4sterbunny @0xlucian @sukanyaparashar please have a look at this PR:

1. Edited Hardhat Example
2. Added Foundry Example
3. Add the foundry example to sidebar.js

### CODE REVIEW

File: docs/developing/connect_rpc.md [hunk 1]
```diff
@@ -4,10 +4,11 @@ proofedDate: 20230526
 iterationBy: na
 includedInSite: true
 approvedBy: na
-comment: 
+comment: todo Find out from Yury whether we can have a NeonLabs youtube domain to support the video material that has been created to complement this page
```
Comment (m4sterbunny) [inline docs/developing/connect_rpc.md:7]: @YYYurchenko I created video material that supports this page. Once it is clear who has mandate for docs pages, if it is still product, then I want to link to that video support

**Comment by m4sterbunny** [comment]
@mnedelchev-vn There are conflicts on this branch since merging other branches PTAL

**Comment by mnedelchev-vn** [comment]
@m4sterbunny the conflict in this branch is made by you, because you've changed file `static/data/link-list.json` which was recently updated in the `main` branch. I guess undo your change and it include it in a separated PR.

**Comment by m4sterbunny** [comment]
@mnedelchev-vn confused! GitHub says that these are the conflicting files:
image

**Comment by m4sterbunny** [comment]
none of which are a json file

**Comment by m4sterbunny** [comment]
@mnedelchev-vn I have fixed the conflicts -- there are 2 items pending with you. PTAL and ping me when done.

**Review by m4sterbunny — CHANGES_REQUESTED** [review]
finish the review idiot

File: docs/developing/deploy_facilities/using_foundry.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+:::info
+This step requires an EVM-compatible wallet such as MetaMask, connected to Neon Devnet, with a balance in Devnet NEON available from [NeonFaucet](https://neonfaucet.org/).
+
+The following tutorials assist you to meet these prerequisites:
+- Learn how to [install a MetaMask wallet and connect it to Devnet](/docs/wallet/metamask_setup)
+> Or [connect an existing wallet to Devnet](/docs/developing/connect_rpc#connect-via-chainlist)
+:::
+
+<!-- I suspect there is no need to create new wallet as per first draft of page >> existing wallet probably fine -- test on run through -->
```
[the remark this answers is not in the document]
Comment (mnedelchev-vn) [reply docs/developing/deploy_facilities/using_foundry.md:66]: @m4sterbunny I'm not happy with the way you cut the example of using foundry. Currently in file [docs/developing/deploy_facilities/using_foundry.md](https://github.com/neonlabsorg/neon-evm.docs/pull/110/files/2765014529675ed3f9dbe47955944a6a8eaffee3#diff-a844f10ba30859fca4e55caa4193c1ece97430b551fdfb0ae3e69c24d768915d) you've left only one step which is the deployment of the contract. We need back:
1. Installation step
2. Setup of the environment file
3. Compile contracts
4. Deploy contracts
5. Verify contracts

Plus the verification with foundry is a lot different than verification with hardhat so it has to be explained here in detail also. If you want to shorten my initial file then remove the MetaMask info from step 2, but anything else is important to have.

[the remark this answers is not in the document]
Comment (m4sterbunny) [reply docs/developing/deploy_facilities/using_foundry.md:66]: @mnedelchev-vn I assume that you are a little git confused -- GitHub does not provide the easiest UX, and you would probably enjoy using the competitors. I have pulled this branch (1) and am looking at the file using_foundry.md (2) I see Step 1 (3) 
image
Step 2
image

Step 3 
image

Step 4
image

Step 5
image

If you need help pulling a branch locally that has been changed remotely -- give me a shout and I can show you how

[the remark this answers is not in the document]
Comment (mnedelchev-vn) [reply docs/developing/deploy_facilities/using_foundry.md:66]: > @m4sterbunny I'm not happy with the way you cut the example of using foundry. Currently in file [docs/developing/deploy_facilities/using_foundry.md](https://github.com/neonlabsorg/neon-evm.docs/pull/110/files/2765014529675ed3f9dbe47955944a6a8eaffee3#diff-a844f10ba30859fca4e55caa4193c1ece97430b551fdfb0ae3e69c24d768915d) you've left only one step which is the deployment of the contract. We need back:
> 
> 1. Installation step
> 2. Setup of the environment file step
> 3. Compile contracts step
> 4. Deploy contracts step
> 5. Verify contracts step
> 
> Plus the verification with foundry is a lot different than verification with hardhat so it has to be explained here in detail also. If you want to shorten my initial file then remove the MetaMask info from step 2, but anything else is important to have.

@m4sterbunny nvm this.

File: docs/developing/deploy_facilities/using_foundry.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+includedInSite: true
+approvedBy: na
+comment: 
+---
+
+import mm_p_key from '@site/static/img/doc-images/developing/deploy_facilities/foundry-metamask.png';
+
+This tutorial walks through deploying a contract describing an ERC-20 token to Neon Devnet. 
+
+<!-- I don't see this step: You test this contract by transferring tokens to randomly generated wallet. todo verify if this can be added, or if I missed it -->
```
[the remark this answers is not in the document]
Comment (mnedelchev-vn) [reply docs/developing/deploy_facilities/using_foundry.md:14]: We don't have this example inside the foundry github repo.

**Comment by mnedelchev-vn** [comment]
@m4sterbunny there was some wrong stuff with `using_hardhat.mdx` file, for whatever reason the content was a real mess. I made a new commit where I recreated the steps to deploy thru hardhat. Please check Hardhat & Foundry pages and lmk if we missed smth.

File: docs/developing/deploy_facilities/using_foundry.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (m4sterbunny) [reply docs/developing/deploy_facilities/using_foundry.md:14]: I am deleting the unsupported step

**Review by m4sterbunny — APPROVED** [review]
I did DIY review fixes as this languished too long

**Pull request merged**